Add change environment variables and image for service

Glowbal opened the pull request:
- Add ability to add / remove and change environment variables of a service.
- Add ability to change the image of a service.
- Change saving of an edited service now happens when pressing save instead of after a change. This is done to let you change multiple values at the same time when updating the service. Also adds a reset option to restore changed values to before they were edited.
- Initial work for #304.

deviantony commented:
I do like the ability to update multiple fields for a service ! Really cool. On a purely UX/design I don't like this big "Service has been edited" banner popping in the middle of the view. IMO this should be displayed either under the service details table (inside the widget) on a single line or in the head of the widget (pulled to the right?). 

I also like the ability to reset to previous state.

deviantony commented:
Waiting for @ncresswell to review.

Glowbal commented:
@deviantony I've changed the location of the save and reset button. I've also fixed an issue with the reset where environment variable changes would not be properly cleared out and updated the placeholder values for the environment variables to be consistent with the create service view.

<img width="868" alt="screen shot 2016-11-03 at 21 47 35" src="https://cloud.githubusercontent.com/assets/5007509/19984607/70896034-a20f-11e6-8576-40d96a18dccb.png">

deviantony commented:
Awesome ! Will review it again soon.

deviantony commented on app/components/service/service.html:
```diff
@@ -66,15 +75,35 @@
+                    <div ng-repeat="var in service.EnvironmentVariables" style="margin-top: 2px;">
+                      <div class="input-group col-sm-5 input-group-sm">
+                        <span class="input-group-addon">name</span>
+                        <input type="text" class="form-control" ng-model="var.key" ng-disabled="var.added" placeholder="e.g. FOO">
```
To be consistent with the service creation view, the placeholder for a var name should be `com.example.foo`

Glowbal replied:
Placeholder for a variable name in the service creation view is `FOO`. `com.example.foo` is used for labels.

deviantony replied:
@Glowbal my bad, thanks !

deviantony commented:
Thanks a lot @Glowbal :)
